docs/addons/jwt.md

```diff
@@ -0,0 +1,315 @@
+
+    ```php
+    // new file - app/Config/AuthJWT.php
+    <?php
+
+    declare(strict_types=1);
+
+    namespace Config;
+
+    use CodeIgniter\Shield\Config\AuthJWT as ShieldAuthJWT;
+
+    /**
+     * JWT Authenticator Configuration
+     */
+    class AuthJWT extends ShieldAuthJWT
+    {
+        // ...
+    }
+    ```
+
+3. If your **app/Config/Auth.php** is not up-to-date, you also need to update it. Check **vendor/codeigniter4/shield/src/Config/Auth.php** and apply the differences.
+
+## Configuration
+
+Configure **app/Config/AuthJWT.php** for your needs.
+
+### Set the Default Claims
+
+> **Note**
+> A payload contains the actual data being transmitted, such as user ID, role,
+> or expiration time. Items in a payload is called *claims*.
+
+Set the default payload items to the property `$defaultClaims`.
+
+E.g.:
+```php
+    public array $defaultClaims = [
+        'iss' => 'https://codeigniter.com/',
+    ];
+```
+
+The default claims will be included in all tokens issued by Shield.
+
+### Set Secret Key
+
+Set your secret key in the `$keys` property, or set it in your `.env` file.
+
+E.g.:
+```dotenv
+authjwt.keys.default.0.secret = 8XBFsF6HThIa7OV/bSynahEch+WbKrGcuiJVYPiwqPE=
+```
+
+It needs at least 256 bits random string. The length of the secret depends on the 
+algorithm we use. The default one is `HS256`, so to ensure that the hash value is 
+secure and not easily guessable, the secret key should be at least as long as the 
+hash function's output - 256 bits (32 bytes). You can get a secure random string 
+with the following command:
+
+```console
+php -r 'echo base64_encode(random_bytes(32));'
+```
+
+> **Note**
+> The secret key is used for signing and validating tokens.
+
+## Issuing JWTs
+
+To use JWT Authentication, you need a controller that issues JWTs.
+
+Here is a sample controller. When a client posts valid credentials (email/password),
+it returns a new JWT.
+
+```php
+// Routes.php
+$routes->post('auth/jwt', '\App\Controllers\Auth\LoginController::jwtLogin');
+
+// LoginController.php
+namespace App\Controllers\Auth;
+
+use App\Controllers\BaseController;
+use CodeIgniter\API\ResponseTrait;
+use CodeIgniter\HTTP\ResponseInterface;
+use CodeIgniter\Shield\Authentication\Authenticators\Session;
+use CodeIgniter\Shield\Authentication\JWTManager;
+use CodeIgniter\Shield\Authentication\Passwords;
+use CodeIgniter\Shield\Config\AuthSession;
+
+class LoginController extends BaseController
+{
+    use ResponseTrait;
+
+    /**
+     * Authenticate Existing User and Issue JWT.
+     */
+    public function jwtLogin(): ResponseInterface
+    {
+        // Get the validation rules
+        $rules = $this->getValidationRules();
+
+        // Validate credentials
+        if (! $this->validateData($this->request->getPost(), $rules)) {
+            return $this->failValidationErrors($this->validator->getErrors(), 422);
```

Comment:
```php
return $this->failValidationErrors($this->validator->getErrors(), 422);
```
Here, 422 cannot be a `int`. Maybe using `setStatusCode(422)` is better.
Considering that I don't know if you intended to set the status code or just a 422 error, I didn't add the change suggestion.

(https://github.com/datamweb/CodeIgniter4/blob/e151d3a25f3ed6a2ff29d022499ed21cd3ade259/system/API/ResponseTrait.php#L238-L248)

Reply:
Thanks! You are correct.

422 came from https://github.com/codeigniter4/shield/blob/779bb12aac715460abde88d84e02bab9776b96ad/docs/guides/mobile_apps.md?plain=1#L44
But it seems the code is not good for.

> [15.5.21. ](https://www.rfc-editor.org/rfc/rfc9110.html#section-15.5.21)[422 Unprocessable Content](https://www.rfc-editor.org/rfc/rfc9110.html#name-422-unprocessable-content)
>
> The 422 (Unprocessable Content) status code indicates that the server understands the content type of the request content (hence a [415 (Unsupported Media Type)](https://www.rfc-editor.org/rfc/rfc9110.html#status.415) status code is inappropriate), and the syntax of the request content is correct, but it was unable to process the contained instructions. For example, this status code can be sent if an XML request content contains well-formed (i.e., syntactically correct), but semantically erroneous XML instructions.
>
> http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
https://www.rfc-editor.org/rfc/rfc9110.html#section-15.5.21

Reply:
Changed the code.

Reply:
> 422 came from

There, too, it seems that it should change.